pandas-stubs/core/indexes/accessors.pyi

```diff
@@ -379,3 +388,32 @@ class DatetimeIndexProperties(
     def std(
         self, axis: int | None = ..., ddof: int = ..., skipna: bool = ...
     ) -> Timedelta: ...
+
+# For some reason, using TimedeltaIndex as an argument to _DatetimeRoundingMethods
+# doesn't work for pyright.  So we just make the rounding methods explicit here.
+class TimedeltaIndexProperties(
+    Properties,
+    _TimedeltaPropertiesNoRounding[Index, Index],
+    # _DatetimeRoundingMethods[TimedeltaIndex],
+):
+    def round(
+        self,
+        freq: str | BaseOffset | None,
+        ambiguous: Literal["raise", "infer", "NaT"] | np_ndarray_bool = ...,
+        nonexistent: Literal["shift_forward", "shift_backward", "NaT", "raise"]
+        | Timedelta = ...,
+    ) -> TimedeltaIndex: ...
+    def floor(
+        self,
+        freq: str | BaseOffset | None,
```

Comment:
Can be in a followup PR: `_typing` in pandas defines `Frequency` as `BaseOffset | str`, could use this here